tests/components/blue_current/test_init.py

```diff
@@ -46,176 +48,162 @@ async def test_load_unload_entry(hass: HomeAssistant) -> None:
+    config_entry: MockConfigEntry,
+    api_error: BlueCurrentException,
+    config_error: IntegrationError,
 ) -> None:
-    """Tests if the correct config error is raised when connecting to the api fails."""
+    """Test if the correct config error is raised when connecting to the api fails."""
     with patch(
         "homeassistant.components.blue_current.Client.connect",
         side_effect=api_error,
     ), pytest.raises(config_error):
-        config_entry = MockConfigEntry(
-            domain=DOMAIN,
-            entry_id="uuid",
-            unique_id="uuid",
-            data={"api_token": "123", "card": {"123"}},
-        )
         config_entry.add_to_hass(hass)
-
         await async_setup_entry(hass, config_entry)
 
 
-async def test_on_data(hass: HomeAssistant) -> None:
+async def test_on_data(hass: HomeAssistant, config_entry: MockConfigEntry) -> None:
     """Test on_data."""
 
-    await init_integration(hass, "sensor", {})
-
-    with patch(
-        "homeassistant.components.blue_current.async_dispatcher_send"
-    ) as test_async_dispatcher_send:
-        connector: Connector = hass.data[DOMAIN]["uuid"]
-
-        # test CHARGE_POINTS
-        data = {
-            "object": "CHARGE_POINTS",
-            "data": [{"evse_id": "101", "model_type": "hidden", "name": ""}],
-        }
-        await connector.on_data(data)
-        assert connector.charge_points == {"101": {"model_type": "hidden", "name": ""}}
-
-        # test CH_STATUS
-        data2 = {
-            "object": "CH_STATUS",
-            "data": {
-                "actual_v1": 12,
-                "actual_v2": 14,
-                "actual_v3": 15,
-                "actual_p1": 12,
-                "actual_p2": 14,
-                "actual_p3": 15,
-                "activity": "charging",
-                "start_datetime": "2021-11-18T14:12:23",
-                "stop_datetime": "2021-11-18T14:32:23",
-                "offline_since": "2021-11-18T14:32:23",
-                "total_cost": 10.52,
-                "vehicle_status": "standby",
-                "actual_kwh": 10,
-                "evse_id": "101",
-            },
-        }
-        await connector.on_data(data2)
-        assert connector.charge_points == {
-            "101": {
-                "model_type": "hidden",
-                "name": "",
-                "actual_v1": 12,
-                "actual_v2": 14,
-                "actual_v3": 15,
-                "actual_p1": 12,
-                "actual_p2": 14,
-                "actual_p3": 15,
-                "activity": "charging",
-                "start_datetime": "2021-11-18T14:12:23",
-                "stop_datetime": "2021-11-18T14:32:23",
-                "offline_since": "2021-11-18T14:32:23",
-                "total_cost": 10.52,
-                "vehicle_status": "standby",
-                "actual_kwh": 10,
-            }
+    mock_client = MagicMock(spec=Client)
+    connector = Connector(hass, config_entry, mock_client)
```

Comment:
Don't use any integration details, like the connector. This test should just test that when the client callbacks with new data, an entity in the state machine should be updated with new state as expected.

The client callback is available as a mock call argument when patching the client, something like `mock_client_class.return_value.start_loop.call_args[0][0]`.

Reply:
Should i not test the connector itself? That is what test_on_data, test_start_loop and test_reconnect_* are for. Testing if the entity state is set correctly is done in test_sensor.py.

Reply:
No, we should not test integration details directly. This allows us to to refactor the integration without changing the tests.

Reply:
Alright, so i think i should remove those tests in test_init.py right? (Since the entity state is tested in test_sensor.py).

Reply:
Yes, if we already test it in the sensor tests we don't need to test it again here.